### ISSUE TITLE
Inlining produces some erroneous results in for loops

### ISSUE THREAD

**Issue by dcjones** [JuliaLang/julia#10993]
Here's an interesting one.

```julia
using Compose

@show 1*cy == cy # true
for i in 1:1
    @show i*cy == cy # false
end
```

The issue occurs in the `i*cy` not the `==`. The value that results is as if `i` was 0 instead of 1.

Yet if I run this with `julia --inline=no` this works as expected. I can try to reduce this example (i.e. remove the `using Compose`) if that's helpful.

**Comment by dcjones** [comment]
Bite sized test case:
```julia
immutable T
    x::Nothing
    y::Int
    z::Int
end

const a = T(nothing, 0, 1)
@show T(nothing, a.y, 1*a.z) == a # true
for i in 1:1
    @show T(nothing, a.y, i*a.z) == a # false, should be true
end
```

**Comment by Keno** [comment]
This was fixed in #11044.

**Comment by tkelman** [comment]
test should still be added?
